patches/minecraft/net/minecraft/world/item/Rarity.java.patch

```diff
@@ -9,13 +9,31 @@
     COMMON(ChatFormatting.WHITE),
     UNCOMMON(ChatFormatting.YELLOW),
     RARE(ChatFormatting.AQUA),
-@@ -12,5 +_,9 @@
+    EPIC(ChatFormatting.LIGHT_PURPLE);
+ 
++   @Deprecated // Forge: Use getStyleModifier
+    public final ChatFormatting f_43022_;
++   private final java.util.function.UnaryOperator<net.minecraft.network.chat.Style> styleModifier;
  
     private Rarity(ChatFormatting p_43028_) {
        this.f_43022_ = p_43028_;
++      this.styleModifier = style -> style.m_131140_(p_43028_);
++   }
++
++   Rarity(java.util.function.UnaryOperator<net.minecraft.network.chat.Style> styleModifier) {
++      this.f_43022_ = ChatFormatting.BLACK;
```

Comment:
i would change this to ChatFormatting.RESET this way anyone who uses it to get the color via ChatFormatting#getColor() will get null

Reply:
Which they might not be prepared for at all.

Reply:
Well the return type of `ChatFormatting#getColor()` is a nullable Integer object, so they always had to be prepared for it to be null 😆 

Anyways i don't really care, `ChatFormatting.BLACK` is acceptable